documentation/docs/libraries/java/getting_started.md

```diff
@@ -7,102 +7,110 @@ keywords:
 
-### Step 1: Creating the native library
+1. JNI Native library linking `Rust` to `C`, and then `C` to java `native` methods (`.so` , `.dll` or `.dylib` depending on your system).
+2. Java archive(Jar) containing `native` methods which call C code. (`.jar`).
+
+### Step 1: Creating the Native Library
+
+Build the wallet library (this generates the java source code and JNI library file):
 
-Build the wallet library (This generates the java source code and JNI library file)
 ```
 cd wallet.rs/bindings/java
 cargo build --release
 ```
 
-Generated binaries can then be found at `wallet.rs/bindings/java/target/release`
+Generated binaries can then be found in `wallet.rs/bindings/java/target/release`.
+
+:::note
 
-:::info
 Compiling for Android requires additional compilation instructions.
-Specific instructions can be found in [Android development](android_development)
+These instructions can be found in the [Android development](android_development) section.
 :::
 
-### Step 2: Creating the Java archive
+### Step 2: Creating the Java Archive
 
 #### Generating the source files and classes
 
-After step 1, Java source files will be generated under `wallet.rs/bindings/java/native/src/main/java/org/iota/wallet`.
+After you complete step 1, Java source files will be generated under `wallet.rs/bindings/java/native/src/main/java/org/iota/wallet`.
 
-If this step succeeded, we need to generate the jar file containing the newly generated Java source files.
+If this step succeeds, you need to generate the jar file containing the newly generated Java source files.
 
 #### Generating the jar
 
-Generating the jar can be done with your tool of preference. We provide examples for Gradle and Maven in this guide
+Generating the jar can be done with your tool of preference. We provide examples for Gradle and Maven in this guide.
 
 ##### Gradle
 
-Make `gradlew` executable (`chmod +x gradlew`) if needed, then run
+Make `gradlew` executable (`chmod +x gradlew`) if needed, then run:
+
 ```
 cd wallet.rs/bindings/java
 ./gradlew jar
 ```
 
 ##### Maven
+
 ```
 cd wallet.rs/bindings/java
 mvn install
 ```
 
 After running one of these commands, the jar can then be found at `wallet.rs/bindings/java/native/build/libs/native.jar`
 
-
 ## Usage
+
 You can find more information on using the `wallet.rs` library's java binding in the [examples section](examples.md).
 
 ### Gradle
+
 ```
 ./gradlew examples:java-app:test --info
 ```
 
 ### Maven
+
 ```
 mvn exec:exec
 ```
 
 ## API Reference
 
-API Reference can be found [here](api_reference).
+You can find the API Reference [here](api_reference).
 
 ## Limitations
 
-Due to the fact that we are linking through C from Rust, there are a couple of limiting factors.
+Due to the fact that we are linking through C from Rust, there are a few limiting factors.
 
 - Classic builder patterns return a `clone` after each builder call since we can only pass back to C by reference in `Rust`
```

Comment:
```suggestion
- Classic builder patterns return a `clone` after each builder call since we can only pass back to C by reference in `Rust` as shown in the example below.
```